node/lib/openshift-origin-node/model/application_container_ext/environment.rb

```diff
@@ -248,8 +248,8 @@ def replace_ssh_keys(ssh_keys)
           krb5_principals = ssh_keys.select {|k| k['type'] == 'krb5-principal'}
 
           self.class.notify_observers(:before_replace_ssh_keys, self)
-          AuthorizedKeysFile.new(self).replace_keys(authorized_keys) if authorized_keys.count > 0
-          K5login.new(self).replace_principals(krb5_principals) if krb5_principals.count > 0
+          AuthorizedKeysFile.new(self).replace_keys(authorized_keys)
+          K5login.new(self).replace_principals(krb5_principals)
```

Comment:
Please add a comment in here that confirms that cleaning out the ssh keys if the particular lists are empty is actually intentional.